packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,460 @@
+# Description
+
+Jobs is the Angular DevKit subsystem for scheduling and running generic functions with clearly 
+typed inputs and outputs. A Job is code (ie. a function) that can be scheduled, synchronized and
+schedule other jobs. Those Jobs can be run in parallel or follow an execution strategy. They can
+also be extended to allow for transformations, combination and refactoring through different 
+operations.
+
+The whole API is serializable, allowing someone to use a Node Stream or message channel to
+communicate between the job and the job scheduler.
+
+Jobs are lazy, cold and guaranteed to be only executed once per scheduling. Subscribing to a job
+will return events from the point where the job is at.
+
+## Argument, Input, Output and Channels
+Jobs receives an argument when first started. They can emit multiple outputs. In addition, jobs can
+listen to an input channel, and provide multiple output channels. Those channels emit asynchronous
+JSON messages that can be typed. In this context, input and output are always in relation to the 
+Job code itself, not the caller.
+
+If a Job was an executable, the argument would be the arguments on the command line, the input 
+channel would be STDIN, the output channel would be STDOUT, and the channels would be additional
+pipes out.
+
+In addition, a Job has a logging channel that can be used to log messages to the original caller.
+These events need to be listened to (or forwarded) by the code scheduling the job. You can think 
+of those events as STDERR.
+
+## LifeCycle
+A Job goes through multiple LifeCycle events before its completion;
+1. `JobState.Queued`. The job was queued and is waiting to start.
+1. `JobState.Created`. The job's dependencies (see
+    ["Synchronizing and Dependencies"](#Dependencies)) are done running, and the job's argument
+    will be validated before the Job's code will be executed.
+1. `JobState.Started`. The job has been started. The job's code is then called.
+1. `JobState.Ended`. The job has ended and is done running.
+
+Each state (except Queued) corresponds to a JobEvent on the `rawOutput` observable that 
+triggers the state change. The `Queued` state is set as the job is scheduled, so there is no need
+to listen to an event. A user can listen to these events, or use the `state` member which will
+correspond to one of those. The Job implementation itself should not emit those events (they 
+will be filtered out); the Scheduler should take care of it.
```

Comment:
...triggers the state change. The Scheduler emits these events; the job implementation should not emit them, and if it does they are filtered out. You can listen for these events or use the corresponding `state` member. The `Queued` state is set as the job is scheduled, so there is no need to listen for the event.